Xamarin.Forms.Core.UnitTests/CarouselViewTests.cs

```diff
@@ -0,0 +1,159 @@
+			Assert.IsTrue(carouselView.ItemSizingStrategy == ItemSizingStrategy.None);
+		}
+
+		[Test]
+		public void TestPositionChangedCommand()
+		{
+			var source = new List<string> { "1", "2", "3" };
+			var carouselView = new CarouselView
+			{
+				ItemsSource = source
+			};
+
+			int countFired = 0;
+			carouselView.PositionChangedCommand = new Command(() =>
+			{
+				countFired = countFired + 1;
+			});
+			Assert.AreSame(source, carouselView.ItemsSource);
+			carouselView.Position = 1;
+			Assert.IsTrue(countFired == 1);
+		}
+
+		public void TestPositionChangedEvent()
+		{
+			var gotoPosition = 1;
+			var source = new List<string> { "1", "2", "3" };
+			var carouselView = new CarouselView
+			{
+				ItemsSource = source
+			};
+
+			int countFired = 0;
+			carouselView.PositionChanged += (s, e) =>
+			{
+				countFired += 1;
+			};
+			Assert.AreSame(source, carouselView.ItemsSource);
+			carouselView.Position = gotoPosition;
+			Assert.IsTrue(countFired == 1);
+		}
+
+		[Test]
+		public void TestCurrentItemChangesPosition()
+		{
+			var gotoPosition = 1;
+			var source = new List<string> { "1", "2", "3" };
+			var carouselView = new CarouselView
+			{
+				ItemsSource = source
+			};
+
+			int countFired = 0;
+			carouselView.PositionChangedCommand = new Command(() =>
+			{
+				countFired += 1;
+			});
+			Assert.AreSame(source, carouselView.ItemsSource);
+			carouselView.CurrentItem = source[gotoPosition];
+			Assert.AreEqual(gotoPosition, carouselView.Position);
+		}
+
+		[Test]
+		public void TestCurrentItemChangedCommand()
+		{
+			var gotoPosition = 1;
+			var source = new List<string> { "1", "2", "3" };
+			var carouselView = new CarouselView
+			{
+				ItemsSource = source
+			};
+
+			int countFired = 0;
+			carouselView.CurrentItemChangedCommand = new Command(() =>
+			{
+				countFired += 1;
+			});
+			Assert.AreSame(source, carouselView.ItemsSource);
+			carouselView.CurrentItem = source[gotoPosition];
+			Assert.IsTrue(countFired == 1);
+		}
+
+		[Test]
+		public void TestCurrentItemChangedEvent()
+		{
+			var gotoPosition = 1;
+			var source = new List<string> { "1", "2", "3" };
+			var carouselView = new CarouselView
+			{
+				ItemsSource = source
+			};
+
+			int countFired = 0;
+			carouselView.CurrentItemChanged += (s, e) =>
+			{
+				countFired += 1;
+			};
+			Assert.AreSame(source, carouselView.ItemsSource);
+			carouselView.CurrentItem = source[gotoPosition];
+			Assert.IsTrue(countFired == 1);
+		}
+
+		[Test]
+		public void TestScrolled()
+		{
+			var scrollPosition = 100;
+			var scrollDirection = ScrollDirection.Down;
+			var scrolledDirection = ScrollDirection.Up;
+			var source = new List<string> { "1", "2", "3" };
+			var carouselView = new CarouselView
+			{
+				ItemsSource = source
+			};
+			int countFired = 0;
+			carouselView.Scrolled += (s, e) =>
+			{
+				countFired += 1;
+				scrolledDirection = e.Direction;
+			};
+			carouselView.SendScrolled(scrollPosition, scrollDirection);
```

Comment:
`'CarouselView' does not contain a definition for 'SendScrolled' and no accessible extension method 'SendScrolled' accepting a first argument of type 'CarouselView' could be found`